app/handlers/tests/test_handlers_resource.py

```diff
@@ -100,6 +91,42 @@ def test_update(self, mock_api_client):
         mock_api_client.resource_update.assert_called_once_with(new_resource_spec)
         assert patch_mock.spec == {}
 
+    def test_update_caused_by_create_does_nothing(self, mock_api_client):
+        rid = "UmVzb3VyY2U6OTMxODE3"
+        spec = new = {
+            "id": rid,
+            "address": "my.default.cluster.local",
+            "name": "new-name",
+        }
+        diff = (("add", ("id",), rid),)
```

Comment:
nit: I think even for `add` action, there is still the old value of `None` ([doc](https://kopf.readthedocs.io/en/stable/walkthrough/diffs/#diffs)) so this should be
```python
diff = (("add", ("id",), None, rid),)
```